src/DBFTPlugin/RecoveryMessage.cs

```diff
@@ -17,32 +17,45 @@ public partial class RecoveryMessage : ConsensusMessage
         public Dictionary<int, PreparationPayloadCompact> PreparationMessages;
         public Dictionary<int, CommitPayloadCompact> CommitMessages;
 
+        private readonly ulong validatorsCount;
+
         public override int Size => base.Size
             + /* ChangeViewMessages */ ChangeViewMessages?.Values.GetVarSize() ?? 0
             + /* PrepareRequestMessage */ 1 + PrepareRequestMessage?.Size ?? 0
             + /* PreparationHash */ PreparationHash?.Size ?? 0
             + /* PreparationMessages */ PreparationMessages?.Values.GetVarSize() ?? 0
             + /* CommitMessages */ CommitMessages?.Values.GetVarSize() ?? 0;
 
-        public RecoveryMessage() : base(ConsensusMessageType.RecoveryMessage)
+        public RecoveryMessage(int validatorsCount) : base(ConsensusMessageType.RecoveryMessage)
```

Comment:
Why do we need `validatorsCount` in the constructor?

Reply:
It's composed for different messages that we need to validate

Reply:
Can we verify it in `Verify()` method?

Reply:
Done.